x-pack/plugins/cases/server/services/alerts/index.ts

```diff
@@ -18,27 +19,200 @@ import {
   ALERT_WORKFLOW_STATUS,
   STATUS_VALUES,
 } from '../../../../rule_registry/common/technical_rule_data_field_names';
-
-interface Alert {
-  _id: string;
-  _index: string;
-  _source: Record<string, unknown>;
-}
-
-interface AlertsResponse {
-  docs: Alert[];
-}
-
-function isEmptyAlert(alert: AlertInfo): boolean {
-  return isEmpty(alert.id) || isEmpty(alert.index);
-}
+import {
+  AggregationFields,
+  FrequencyResult,
+  HostAggregate,
+  UniqueCountResult,
+  UserAggregate,
+} from './types';
 
 export class AlertService {
   constructor(
     private readonly scopedClusterClient: ElasticsearchClient,
     private readonly logger: Logger
   ) {}
 
+  public async countUniqueValuesForFields({
+    fields,
+    alerts,
+  }: {
+    fields: AggregationFields[];
+    alerts: AlertIdIndex[];
+  }): Promise<UniqueCountResult> {
+    try {
+      const { ids, indices } = AlertService.getUniqueIdsIndices(alerts);
+
+      const res = await this.scopedClusterClient.search({
+        index: indices,
+        query: { ids: { values: ids } },
+        size: 0,
+        aggregations: AlertService.buildTotalUniqueAggregations(fields),
+      });
+
+      const aggs = res.body.aggregations as UniqueFieldAggregate;
+
+      return {
+        totalHosts: aggs.hosts?.value,
+        totalUsers: aggs.users?.value,
+      };
+    } catch (error) {
+      throw createCaseError({
+        message: `Failed to count totals for fields: ${JSON.stringify(fields)}: ${error}`,
+        error,
+        logger: this.logger,
+      });
+    }
+  }
+
+  private static buildTotalUniqueAggregations(fields: AggregationFields[]) {
+    return fields.reduce<Record<string, estypes.AggregationsAggregationContainer>>((acc, field) => {
+      switch (field) {
```

Comment:
I think the `AlertingService` should be agnostic of the fields. We can build the query dynamically by iterating the `fields` argument. You may need more information from the `AggregationFields`.